community/contributing/best_practices_for_engine_contributors.rst

```diff
@@ -69,9 +68,9 @@ its characterized by lines of thought such as:
 - I think users will eventually need to...
 
 This is generally considered a bad habit, because trying to solve problems that
-*don't actually exist* in the present will very often lead to code that will be
-written but never used, or to code that is considerably more complex to use and
-maintain than it needs to be.
+*don't actually exist* in the present will very often lead to the code that will
+be written but never used, or the code that is considerably more complex to use
```

Comment:
The sentence seems weird to me with the added "the" before code.

"lead to (written, but unused) code", not "lead to the written but unused code" (which code is "the code"? It's generic).

Reply:
In the previous "to code that is considerably more complex", "code" becomes a verb, so that was quite confusing for me. I'm not particularly strong with those English grammatical articles such as `the` or `a` (they don't exist in my language at all), but according to definition of `the` I've googled:

> denoting one or more people or things **already mentioned or assumed to be common knowledge** [emphasis mine].

Best practices could be considered common knowledge so that's why I added those, "*the* code which is...", rather than "to code", because "to" was already there in "will very often lead to".

Reply:
> In the previous "to code that is considerably more complex", "code" becomes a verb, so that was quite confusing for me. 

I don't think that's correct.

See https://dictionary.cambridge.org/dictionary/essential-british-english/lead-to-something

If code were used as a verb here, I think the correct phrasing would be "lead to coding".
Here, "lead to code" means "lead to (the presence of) code".

CC @cbscribe

Reply:
I missed it, code is still a noun in that case, even without the article. Note the page could still use a whole editing pass to simplify the language, like a fair part of the best practices series. The style is a bit heavy.